### DISCUSSION TITLE
Geometry stream to point cloud

### DISCUSSION THREAD

**Discussion by johnnyd710** [iTwin/itwinjs-core#2218] [General] [answered]
Is there a way with imodeljs to transform the geometry (or geometry stream) of an object in an imodel to a point cloud representation of it? So to go from a geometry stream to an array of 3d points ?

**Comment by pmconne** [comment, marked as answer]
No. That sounds like a very niche thing to want to do - not something that would be part of the iTwin.js libraries.
You can certainly use iTwin.js to read the geometry streams of the elements of interest and do whatever you want with them (excluding, for now, BReps and text).

**Reply by bbastings** [reply]
Perhaps IModelDb.exportGraphics would be helpful here? You can supply facetting options to control the meshes returned for any surface or solid in the geometry stream, then use the mesh vertices to construct the point cloud?

**Reply by mgooding** [reply]
Yes, that's what I'd suggest. Get meshes from exportGraphics and feed them into something like https://pointclouds.org/ to convert to point clouds.

**Reply by johnnyd710** [reply]
thank you for the response! good to know about export graphics, I'll look into it.
